crates/optimism/chainspec/src/lib.rs

```diff
@@ -421,6 +422,31 @@ impl OpGenesisInfo {
     }
 }
 
+/// Helper method building a [`Header`] given [`Genesis`] and [`ChainHardforks`].
+pub fn make_op_genesis_header(genesis: &Genesis, hardforks: &ChainHardforks) -> Header {
+    let mut header = reth_chainspec::make_genesis_header(genesis, hardforks);
+
+    // If Isthmus is active, overwrite the withdrawals root with the storage root of predeploy
+    // `L2ToL1MessagePasser.sol`
+    if hardforks.fork(OpHardfork::Isthmus).active_at_timestamp(genesis.timestamp) {
+        match genesis.alloc.get(&ADDRESS_L2_TO_L1_MESSAGE_PASSER) {
+            Some(predeploy) => {
+                if let Some(ref storage) = predeploy.storage {
+                    header.withdrawals_root =
+                        Some(storage_root_unhashed(storage.iter().map(|(k, v)| (*k, (*v).into()))))
+                }
+            }
+            None =>
+                /*debug!(target: "reth::cli",
+                "Isthmus active but predeploy L2ToL1MessagePasser.sol not found in genesis alloc"
+            ),*/
+                {}
```

Comment:
should we use empty withdrawalsroot here?

Reply:
empty withdrawals root has already been set by delegated call to inner type. when we have https://github.com/paradigmxyz/reth/issues/14526 I want to log this helpful message here.